pkg/cli/client_url.go

```diff
@@ -324,12 +336,16 @@ func (cliCtx *cliContext) makeClientConnURL() (url.URL, error) {
 		opts[k] = v
 	}
 
-	userName := cliCtx.sqlConnUser
-	if userName == "" {
-		userName = security.RootUser
+	var err error
+	if netHost != "" {
+		// Only add TLS parameters when using a network connection.
+		userName := cliCtx.sqlConnUser
+		if userName == "" {
+			userName = security.RootUser
+		}
+		err = cliCtx.LoadSecurityOptions(opts, userName)
```

Comment:
minor nit: not usually a fan of err's hiding in nested if statements and returning on the outside. can we just do the err check here and return nil below?

(that said we have a linter for this so could flip either way)